temporal-sdk/src/main/java/io/temporal/internal/worker/ActivityPollTask.java

```diff
@@ -86,31 +89,39 @@ public PollActivityTaskQueueResponse poll() {
     if (log.isTraceEnabled()) {
       log.trace("poll request begin: " + pollRequest);
     }
-    PollActivityTaskQueueResponse result;
+    PollActivityTaskQueueResponse response;
+    boolean isSuccessful = false;
     try {
-      result =
-          service
-              .blockingStub()
-              .withOption(METRICS_TAGS_CALL_OPTIONS_KEY, metricsScope)
-              .pollActivityTaskQueue(pollRequest.build());
-    } catch (StatusRuntimeException e) {
-      if (e.getStatus().getCode() == Status.Code.UNAVAILABLE
-          && e.getMessage().startsWith("UNAVAILABLE: Channel shutdown")) {
-        return null;
+      pollSemaphore.acquire();
```

Comment:
I believe we have to move this acquire to before the try

Reply:
done